openassessment/xblock/staff_area_mixin.py

```diff
@@ -492,9 +492,36 @@ def reschedule_unfinished_tasks(self, data, suffix=''):  # pylint: disable=W0613
                 'msg': self._(u"An error occurred while rescheduling tasks: {}".format(ex))
             }
 
+    def clear_submission(self, user_id, course_id, item_id):
+        """
+            This method is called to reset student state for a given problem. It will
+            cancel the workflow using traditional methods, to remove it from the grading pools,
+            and pass through to the submissions API to orphan the submission so the user can create a new one.
+        """
+        # Note that student_item cannot be constructed using get_student_item_dict, since we're in a staff context
+        student_item = {
+            'course_id': course_id,
+            'student_id': user_id,
+            'item_id': item_id,
+            'item_type': 'openassessment',
+        }
+        # There *should* only be one submission, but the logic is easy to extend for multiples so we may as well do it
+        submissions = submission_api.get_submissions(student_item)
+        for sub in submissions:
+            # Remove the submission from grading pools
+            self._cancel_workflow(sub['uuid'], "Student state cleared", user_id)
```

Comment:
The course author doing the cancellation should be passed (for "cancelled by" instead) instead of the ID corresponding to the learner whose submission is being cancelled.